test/test-iconv-helper.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © 2015 Red Hat, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the next
+ * paragraph) shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include <config.h>
+#include <check.h>
+#include <string.h>
+#include <errno.h>
+
+#include "libratbag-util.h"
+
+START_TEST(iconv_convert_to_utf16le)
+{
+	static const char sample_utf16le[] = { 'F', '\0', 'o', '\0', 'o',
+		'\0' };
+	char output[sizeof(sample_utf16le)];
+	ssize_t rc;
+
+	rc = ratbag_utf8_to_enc(output, sizeof(output), "UTF-16LE", "%s",
+				"Foo");
+
+	ck_assert_int_eq(rc, sizeof(output));
+	ck_assert(memcmp(output, sample_utf16le, sizeof(sample_utf16le)) == 0);
+}
+END_TEST
+
+START_TEST(iconv_convert_from_utf16le)
+{
+	char sample_utf16le[] = { 'F', '\0', 'o', '\0', 'o',
+		'\0' };
+	char *output;
+	ssize_t rc;
+
+	rc = ratbag_utf8_from_enc(sample_utf16le, sizeof(sample_utf16le),
+				  "UTF-16LE", &output);
+
+	ck_assert_int_eq(rc, sizeof("Foo"));
+	ck_assert(memcmp(output, "Foo", sizeof("Foo")) == 0);
+
+	free(output);
+}
+END_TEST
+
+START_TEST(iconv_invalid_encoding)
+{
+	char output[10] = { 0 };
+	ssize_t rc;
+
+	rc = ratbag_utf8_to_enc(output, sizeof(output), "Enterprise UTF-16LE",
+				"%s", "Foo");
+
+	ck_assert_int_eq(rc, -EINVAL);
+}
+END_TEST
+
```

Comment:
Sorry for the duplicate, still getting the hang of this thing…
If you're talking about something like

```c
ratbag_utf8_to_enc(output, sizeof(output), "UTF-16LE", "%s", "F\0\0\0oo");
```
nothing really interesting would happen. `vsnprintf()` would see the first `'\0'` and assume the entirety of the string is `"F"`